packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/Converters.kt

```diff
@@ -50,8 +56,8 @@ import kotlin.reflect.KClass
  * main abstraction of conversion used throughout the library.
  */
 internal interface RealmValueConverter<T> {
-    public fun publicToRealmValue(value: T?): RealmValue
-    public fun realmValueToPublic(realmValue: RealmValue): T?
+    fun MemAllocator.publicToRealmValue(value: T?): RealmValue
```

Comment:
There is something wrong here, because we cannot convert arbitrary arguments in a `MemAllocator`. Strings and binary data would need a `MemTrackingAllocator`! This could be one of the reasons why it would make sense not to have the allocators at this level but somehow burry it in the RealmInterop ... even though we would have to differentiate the paths there ... and then the differentiation would maybe somehow leak back to here anyway 🤔 so maybe not doable.

Reply:
But with the above you would get a runtime error if you use the wrong allocator!! If there is not that much of a difference of using the tracking allocator as you hinted in yesterdays benchmarking discussion, then I would suggest making the typesystem safe and then postpone the optimization to a later PR.